lib/internal/url.js

```diff
@@ -1084,26 +1084,32 @@ function domainToUnicode(domain) {
   return _domainToUnicode(`${domain}`);
 }
 
-// Utility function that converts a URL object into an ordinary
-// options object as expected by the http.request and https.request
-// APIs.
+/**
+ * Utility function that converts a URL object into an ordinary options object
+ * as expected by the `http.request` and `https.request` APIs.
+ * @param {URL} url
```

Comment:
This is wrong. This should be a `URL` or the legacy `Url`

Reply:
It's not how it's documented: https://github.com/nodejs/node/blob/c1178c53ecfdc3fa800d8c1eac962c9b99e22511/doc/api/url.md#L1231

Reply:
If you remove `hostname &&` check, you'll see that there are tests that are failing